heron/scheduler-core/src/java/com/twitter/heron/scheduler/ScalableScheduler.java

```diff
@@ -0,0 +1,36 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.twitter.heron.scheduler;
+
+import java.util.Set;
+
+import com.twitter.heron.spi.packing.PackingPlan;
+
+public interface ScalableScheduler {
+  /**
+   * Requests new containers for scaling a topology
+   *
+   * @param containers Set of containers to be added by the scheduler
+   */
+  void addContainers(Set<PackingPlan.ContainerPlan> containers);
+
+  /**
+   * Requests containers to be released for down-scaling a topology
+   *
+   * @param existingContainers Set of containers currently managed by the scheduler
+   * @param containersToRemove Set of containers to be remove by the scheduler
+   */
+  void removeContainers(Set<PackingPlan.ContainerPlan> existingContainers,
```

Comment:
I was also mulling over that inconsistency. We first implemented as this:
```
void removeContainers(int existingContainerCount, int numContainersToRemove);
```
and for the Aurora impl we could do the math to determine the ids last of the containers to be removed. But Yarn needed to know specifically which ones to remove, so we changed it to the current signature. There was also some thinking that we could assert that `existingContainers` passed was the same as the implementation was already managing, as a type of guard against race conditions of multiple edits.

As I think more about it I don't think either of those reasons are totally valid and we should probably remove `Set existingContainers` from the signature. @ashvina thoughts on that?